### DISCUSSION TITLE
Build Acceleration in VS 17.5

### DISCUSSION THREAD

**Discussion by drewnoakes** [dotnet/project-system#8871] [Announcements]
Now that VS 17.5 is available, it's a great time to try out Build Acceleration in your solution.

Build Acceleration is an opt-in feature that can significantly improve builds times, especially for large solutions.

For the following solutions we saw improvements of around 80%. 

![image](https://user-images.githubusercontent.com/350947/221061855-c2f2f039-88be-41d4-8c8f-8154bf8132e2.png)

We've even had a report of a [98% reduction](https://github.com/dotnet/project-system/issues/8865#issuecomment-1441746389)!

For more information about the feature, and instructions on how to opt-in, see https://aka.ms/vs-build-acceleration

If you have any questions or issues, feel free to comment here or open an issue on this repo's issue tracker.

> ⚠️ [Take the Build Accleration survey!](https://www.surveymonkey.com/r/ZTR6KDX)

**Comment by rodrigoporcionato** [comment]
I don't see the difference in my case. Maybe I do something wrong. I have more than 30 projects and run 4.8 framework. Do It should be working?

**Reply by drewnoakes** [reply]
Please review the requirements listed here https://github.com/dotnet/project-system/blob/main/docs/build-acceleration.md#validate-builds-are-accelerated

From what you describe, it may be that you're not using SDK-style projects. If you are, perhaps you are not using reference assemblies. That document details the steps you can take to enable logging and to diagnose issues. Let me know if anything's unclear and we will improve the docs. Thanks!

**Reply by MaceWindu** [reply]
Enabled verbose logging and tested https://github.com/linq2db/linq2db/pull/3989 build with both enabled and disabled option and don't see any difference and any mention of `accel` substring in build logs

**Reply by MaceWindu** [reply]
But works with another solution
`
Visual Studio accelerated 18 project(s), copying 45 file(s)
`

**Reply by drewnoakes** [reply]
@MaceWindu I pulled down your linq2db PR to try it locally. You're hitting an issue called out in the docs around `ProduceReferenceAssembly`. In 17.6, building your PR produces messages like this:

> FastUpToDate: This project has enabled build acceleration, but at least one referenced project does not produce reference assemblies. Ensure all referenced projects, both direct and indirect, have the 'ProduceReferenceAssembly' MSBuild property set to 'true'. See https://aka.ms/vs-build-acceleration.

It would be a lot more helpful if we included the name of the offending project(s). I filed #8880 to track this.

Adding this property to your projects gets build acceleration working:

```
Build started...
1>FastUpToDate: Input Compile item 'C:\dev\linq2db\Source\LinqToDB\DataContext.cs' is newer (2023-02-28 19:38:26.226) than earliest output 'C:\dev\linq2db\Source\LinqToDB\bin\Debug\net46\linq2db.xml' (2023-02-28 19:38:11.253), not up-to-date. (LinqToDB)
1>------ Build started: Project: LinqToDB, Configuration: Debug Any CPU ------
1>LinqToDB -> C:\dev\linq2db\Source\LinqToDB\bin\Debug\net46\linq2db.dll
1>LinqToDB -> C:\dev\linq2db\Source\LinqToDB\bin\Debug\net472\linq2db.dll
1>LinqToDB -> C:\dev\linq2db\Source\LinqToDB\bin\Debug\netstandard2.0\linq2db.dll
1>LinqToDB -> C:\dev\linq2db\Source\LinqToDB\bin\Debug\netcoreapp3.1\linq2db.dll
1>LinqToDB -> C:\dev\linq2db\Source\LinqToDB\bin\Debug\net7.0\linq2db.dll
1>LinqToDB -> C:\dev\linq2db\Source\LinqToDB\bin\Debug\netstandard2.1\linq2db.dll
1>LinqToDB -> C:\dev\linq2db\Source\LinqToDB\bin\Debug\net6.0\linq2db.dll
========== Build: 1 succeeded, 0 failed, 13 up-to-date, 0 skipped ==========
========== Build started at 7:38 PM and took 14.590 seconds ==========
Visual Studio accelerated 13 project(s), copying 213 file(s). See https://aka.ms/vs-build-acceleration.
```

I'll comment on your PR with the details of what's I had to add.

**Reply by MaceWindu** [reply]
Thanks! Strange I wasn't able to find this message in build logs, maybe did something wrong

**Reply by drewnoakes** [reply]
No, you didn't do anything wrong. The message is not in 17.5. It's something we're adding in 17.6 based on feedback such as yours.

**Comment by mstv** [comment]
How to reset the Build Acceleration cache?

**Reply by drewnoakes** [reply]
There isn't such a cache for build acceleration. What is the scenario you're trying to address?

**Reply by mstv** [reply]
I want to trigger a clean full rebuild. Where are the cached build artifacts stored?

**Reply by drewnoakes** [reply]
> I want to trigger a clean full rebuild. 

Does rebuild do what you want? If not, can you open an issue with more details so we can investigate?

> Where are the cached build artifacts stored?

Everything's in the `obj` and `bin` folders.

**Reply by mstv** [reply]
Thank you for the clarification. I got confused by the "restored" messages on rebuild - like:
```
Restored D:\Build\gitextensions3_dev\GitCommands\GitCommands.csproj (in 41 ms).
```

**Comment by govert** [comment]
The default setting for build acceleration changed in VS 17.9, causing us some trouble, as I explain here: https://github.com/dotnet/project-system/issues/9106#issuecomment-1978464349
